src/utils/generate-type-from-schema.ts

```
@@ -2,19 +2,27 @@ import { compileFromFile } from 'json-schema-to-typescript'
 import fs from 'fs'
 import R from 'ramda'
 
+import { Parameters } from '../params'
+
 // json-schema-to-typescript uses a `toSafeString(string)` function https://github.com/bcherny/json-schema-to-typescript/blob/f41945f19b68918e9c13885f345cb708e1d9898a/src/utils.ts#L163) to obtain a normalized string. This pascalCase mimics this functionality and should address most cases.
 export const pascalCase = (input: string): string =>
   input
     .split(/[^a-zA-Z0-9]+/g)
     .map(word => word.charAt(0).toUpperCase() + word.slice(1))
     .join('')
 
-export const generateTypeFromSchema = async (
-  filePath: string
-): Promise<void> => {
-  const baseTypes = await compileFromFile(filePath)
+export const generateTypeFromSchema = async ({
+  runtimeEnvName,
+  schemaPath,
+  types,
+}: Parameters): Promise<void> => {
+  if (types === false) {
+    return
```

In this case, this Promise technically should be rejected. I'd recommend throwing an error here so that the caller of this function can know to do something practical in the case that no types were generated.

---

Hm not sure I agree with throwing. Not generating types is supported and not an error case.

Does it make sense to move this `if` up to the caller? So `generateTypeFromSchema(..., types)` assumes valid non-falsy `types`

---

I would argue that this _is_ an error case for this function. If `generateTypeFromSchema` cannot generate a type from a schema because it has no types it's an error instead of a success.

---

If the calling function doesn't mind that generating the types errored, then the error can be ignored there.

---

That's why I suggested to move the if-statement up. So that `generateTypeFromSchema` is only called when we generate types for sure.

---

Would agree with both here. Inside this function a non-existant schema is an error IMO. So we should ideally also not trying to call it without one.

---

updated as discussed 👍